hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/StorageContainerServiceProviderImpl.java

```diff
@@ -198,14 +198,18 @@ connectionFactory, getScmDBSnapshotUrl(),
                 ScmConfigKeys.OZONE_SCM_GRPC_PORT_KEY,
                 ScmConfigKeys.OZONE_SCM_GRPC_PORT_DEFAULT);
 
-            try (SCMSnapshotDownloader downloadClient = new InterSCMGrpcClient(
-                hostAddress, grpcPort, configuration,
-                new ReconCertificateClient(new SecurityConfig(configuration),
-                    reconStorage, null, null))) {
-              downloadClient.download(targetFile.toPath()).get();
-            } catch (ExecutionException | InterruptedException e) {
-              LOG.error("Rocks DB checkpoint downloading failed", e);
-              throw new IOException(e);
+            SecurityConfig secConf = new SecurityConfig(configuration);
+            try (ReconCertificateClient certClient =
+                     new ReconCertificateClient(
+                         secConf, reconStorage, null, null)) {
+              try (
```

Comment:
nit: I'm fine with the nested try catch block, but is there an advantage to it compared to the try with multiple resources version? Either way is ok to me.

Reply:
Yes, nice catch, and you are right, there is no reason to keep these two in separate try-with-resources constructs, I just blindly applied the solution to add a try ( before the client creation and the rest, and did not noticed the internal try-with-resources. Fixing this one as well shortly.